react-hyf/react-hyf-client/src/Components/HomeworkWeek3/HomeworkWeek3.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react';
+import NameSurname from './NameSurname';
+import Container from './Container';
+import Info from './Info';
+import Image from './Image';
+import RandomButton from './RandomButton';
+
+class HomeworkWeek3 extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: null,
+      isLoaded: false,
+      items: []
+    };
+  }
+
+  componentDidMount() {
+    fetch('https://uinames.com/api/?ext')
+      .then(res => res.json())
+      .then(
+        result => {
+          this.setState({
+            isLoaded: true,
+            items: result
+          });
+          console.log(this.state.items);
+        },
+        // Note: it's important to handle errors here
+        // instead of a catch() block so that we don't swallow
+        // exceptions from actual bugs in components.
+        error => {
+          this.setState({
+            isLoaded: true,
+            error
+          });
+        }
+      );
+  }
+
+  onClickButton = () => {
+    fetch('https://uinames.com/api/?ext')
+      .then(res => res.json())
+      .then(
+        result => {
+          this.setState({
+            isLoaded: true,
+            items: result
+          });
+          // console.log(this.state.items);
+        },
+        error => {
+          this.setState({
+            isLoaded: true,
+            error
+          });
+        }
+      );
+  };
```

Comment:
I'm seeing some code repetition with the code in `componentDidMount`. This can be easily abstracted into a common function.